shotover-proxy/src/frame/mod.rs

```diff
@@ -14,16 +14,61 @@ pub enum MessageType {
     Kafka,
 }
 
+impl From<&ProtocolType> for MessageType {
+    fn from(value: &ProtocolType) -> Self {
+        match value {
+            ProtocolType::Cassandra { .. } => Self::Cassandra,
+            ProtocolType::Redis => Self::Redis,
+            ProtocolType::Kafka => Self::Kafka,
+        }
+    }
+}
+
+impl Frame {
+    pub fn as_codec_state(&self) -> CodecState {
+        match self {
+            Frame::Cassandra(_) => CodecState::Cassandra {
+                compression: Compression::None,
+            },
+            Frame::Redis(_) => CodecState::Redis,
+        }
+    }
+}
+
 #[derive(PartialEq, Debug, Clone)]
 pub enum Frame {
     Cassandra(CassandraFrame),
     Redis(RedisFrame),
 }
 
 impl Frame {
-    pub fn from_bytes(bytes: Bytes, message_type: MessageType) -> Result<Self> {
+    pub fn from_bytes(
+        bytes: Bytes,
+        message_type: MessageType,
```

Comment:
Can we take a ProtocolType to avoid duplicating these methods?

Reply:
I think we can just remove `from_bytes_with_compression`

Reply:
yep, sounds good!